packages/commercetools/api-client/src/types/Api.ts

```diff
@@ -7,9 +7,21 @@ export interface CustomQuery<T> {
   variables: T;
 }
 
-type CustomQueryFn<T = any> = (query?: any, variables?: T) => CustomQuery<T>
+export type CustomQueryFn<T = any> = (query?: any, variables?: T) => CustomQuery<T>;
+export type CartCustomQueryFn<T> = (
+  cart?: CustomQuery<T>,
+  user?: CustomQuery<T>
+) => { cart: CustomQuery<T>; user: CustomQuery<T> };
 
-export const getCustomQuery = <T = any>(customQueryFn: CustomQueryFn<T>, defaultQuery) => customQueryFn ? customQueryFn() : { query: defaultQuery, variables: {} };
+export const getCustomQuery = <T = any>(customQueryFn: CustomQueryFn<T>, defaultQuery) =>
+  customQueryFn ? customQueryFn() : { query: defaultQuery, variables: {} };
+export const getCartCustomQuery = <T = any>(customQueryFn: CartCustomQueryFn<T>, customQueries) =>
+  customQueryFn
+    ? customQueryFn()
```

Comment:
You have to pass to this function the default queries as sometimes you would need to override variables or only one query.

```
cutomQuery({ cart, user }) => {
  // cart and user  has default queries here
  return {
    cart,
    user: { query: user.query, variables: { I'm changing just variables}  }
  }
} 
```